tasks.py

```diff
@@ -231,6 +233,60 @@ def dev_env(ctx, architecture="amd64", name="kind", cni=None):
         tmp.flush()
         run("kubectl apply -f {}".format(tmp.name), echo=True)
 
+    if protocol == "bgp":
+        print("Configuring MetalLB with a BGP test environment")
+        bgp_dev_env()
+    else:
+        print("Leaving MetalLB unconfigured")
+
+
+# Configure MetalLB in the dev-env for BGP testing.  Start an frr based BGP
+# router in a container and configure MetalLB to peer with it.
+# See dev-env/bgp/README.md for some more information.
+def bgp_dev_env():
+    dev_env_dir = os.getcwd() + "/dev-env/bgp"
+
+    # TODO -- The IP address handling will need updates to add support for IPv6
+
+    node1_ip = run('docker inspect -f "{{ '
+            '.NetworkSettings.Networks.kind.IPAddress }}" kind-control-plane',
+            echo=True)
+    node2_ip = run('docker inspect -f "{{ '
+            '.NetworkSettings.Networks.kind.IPAddress }}" kind-worker',
+            echo=True)
+    node3_ip = run('docker inspect -f "{{ '
+            '.NetworkSettings.Networks.kind.IPAddress }}" kind-worker2',
+            echo=True)
+
+    with open("%s/frr/bgpd.conf.tmpl" % dev_env_dir, 'r') as f:
+        bgpd_config = "! THIS FILE IS AUTOGENERATED\n" + f.read()
+    bgpd_config = bgpd_config.replace("NODE1_IP", node1_ip.stdout.strip())
+    bgpd_config = bgpd_config.replace("NODE2_IP", node2_ip.stdout.strip())
+    bgpd_config = bgpd_config.replace("NODE3_IP", node3_ip.stdout.strip())
+    with open("%s/frr/bgpd.conf" % dev_env_dir, 'w') as f:
+        f.write(bgpd_config)
+
+    # Run a BGP router in a container for all of the speakers to peer with.
+    run("docker rm etcfrr || true", echo=True)
+    run("docker create -v /etc/frr --name etcfrr alpine:3.4 /bin/true",
```

Comment:
Not sure I follow, the `etcfrr` container is created but never run later?

I guess this is needed for the config to be used [here][link]? Just curious, any reason to create this container and use `--volume-from` and not create a directory and use that as volume?

[link]: https://github.com/metallb/metallb/pull/687/files#diff-dc31105b816206de63c47bf42ec2a4d7R277

Reply:
I'm dropping `etcfrr` in the next revision

Reply:
Ohh, permissions! Okay, that seems like a good reason. Maybe we can add a comment for that so it is obvious for other readers?

Reply:
I dropped it already in favor of a temporary dir with the rendered config files.  that works ok

Reply:
Thanks!